sphinx_es_suggest/_static/js/sphinx_es_suggest.js

```diff
@@ -1,4 +1,443 @@
+    // the user.
+    removeResults();
+    let search_loding = document.createElement("div");
+    search_loding.className = "search__result__box";
+    search_loding.innerHTML = "Searching ....";
+    search_outer.appendChild(search_loding);
+
+    fetch(search_url)
+        .then(resp => resp.json())
+        .then(data => {
+            if (data.results.length > 0) {
+                total_results =
+                    MAX_SUGGESTIONS <= data.results.length
+                        ? MAX_SUGGESTIONS
+                        : data.results.length;
+                let search_result_box = generateSuggestionsList(
+                    data,
+                    projectName
+                );
+                removeResults();
+                search_outer.appendChild(search_result_box);
+
+                // remove active classes from all suggestions
+                // if the mouse hovers, otherwise styles from
+                // :hover and .active will clash.
+                search_outer.addEventListener("mouseenter", e => {
+                    removeAllActive();
+                });
+            } else {
+                removeResults();
+                var err_div = getErrorDiv("No Results Found");
+                search_outer.appendChild(err_div);
+            }
+        })
+        .catch(error => {
+            removeResults();
+            var err_div = getErrorDiv("Error Occurred. Please try again.");
+            search_outer.appendChild(err_div);
+            console.log("Search Error (RTD)", error);
+        });
+};
+
+/**
+ * Creates the initial html structure which will be
+ * appended to the <body> node as soon as the page loads.
+ * This html structure will serve as the boilerplate
+ * to show our search results.
+ * It generates the following html structure :-
+ *
+ *  <div class="search__outer__wrapper search__backdrop">
+ *      <div class="search__outer">
+ *          <div class="search__cross">
+ *              <img
+ *                  class="search__cross_img"
+ *                  alt="Close"
+ *                  src="https://image.flaticon.com/icons/svg/59/59836.svg' />"
+ *              />
+ *              <input class="search__outer__input" placeholder="Search ...">
+ *          </div>
+ *      </div>
+ *  </div>
+ *
+ * @return {Object} object containing the nodes with classes
+ *                  "search__outer__wrapper", "search__outer__input" and "search__outer"
+ */
+const generateAndReturnInitialHtml = () => {
+    let search_outer_wrapper = document.createElement("div");
+    search_outer_wrapper.className = "search__outer__wrapper search__backdrop";
+
+    let search_outer = document.createElement("div");
+    search_outer.className = "search__outer";
+
+    let cross_icon = document.createElement("div");
+    cross_icon.className = "search__cross";
+    cross_icon.innerHTML =
+        "<img class='search__cross__img' alt='Close' src='https://image.flaticon.com/icons/svg/59/59836.svg' />";
```

Comment:
Rather than linking to an external image which won't work offline and could be fragile if it was ever (re)moved, I would package it in the repository or inline it.